site/pages/api/admin/resend-invite.api.ts

```diff
@@ -0,0 +1,48 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { USER_MANAGEMENT_PAGE_PATH } from "../../../constants";
+import { createUser, deleteUser as deleteCognitoUser, getUserDetails } from "../../../data/cognito";
+import { user } from "../../../schemas/user-management.schema";
+import { redirectTo, redirectToError } from "../../../utils/apiUtils";
+import { getSession } from "../../../utils/apiUtils/auth";
+
+export interface ResendUserApiRequest extends NextApiRequest {
+    body: {
+        username: string;
+        group: string;
+    };
+}
+
+const resendInvite = async (req: ResendUserApiRequest, res: NextApiResponse): Promise<void> => {
+    try {
+        const { username, group } = req.body;
+
+        const session = getSession(req);
+        if ((session && !session.orgId) || !session || !username || !group) {
+            throw Error("Insufficient values provided to resend an invite");
+        }
+
+        const userDetails = await getUserDetails(username);
+
+        const validatedBody = user.safeParse({ ...userDetails, group });
+
+        if (!validatedBody.success) {
+            throw Error("Insufficient values provided to resend an invite");
+        }
+
+        if (validatedBody.data.orgId !== session.orgId) {
```

Comment:
We should also verify that the user is in a pending status so that we don't delete confirmed users, the status should be "FORCE_CHANGE_PASSWORD"

Reply:
done